### PR TITLE
chore: Add Kotlin Consumer and Driver sample apps.

### PR DESCRIPTION
**Pull request by shawnlu-g**
Add Kotlin Consumer and Driver sample apps.

### CODE REVIEW

**Comment by jpoehnelt** [comment]
@shawnlu-g Please take a moment to fill out [this short survey](https://forms.gle/wYuzN6UibnFj7ZL29). Thank you!

_This is an automated message, feel free to ignore._

**Comment by ArturoSalazarB16** [comment]
Minor thing:
Should we split out a separate PR with changes outside of the kotlin apps?

**Comment by shawnlu-g** [comment]
All changes in this PR are required to make Kotlin sample apps compile and run, a separate PR will be created for additional clean up work e.g. further consolidate dependencies.

**Review by ArturoSalazarB16 — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -4,6 +4,16 @@ This repository contains the source code of the following samples
 
 1. Driver SDK sample (written in Java) in `/java/driver` directory.
 2. Consumer SDK sample (written in Java) in `/java/consumer` directory.
+3. Driver SDK sample (written in Kotlin) in `/kotlin/kotlin-driver` directory.
+4. Consumer SDK sample (written in Kotlin) in `/kotlin/kotlin-consumer` directory.
+
+Note: compared to the Java source, the Kotlin source will have
```
Comment (sukolsak) [inline README.md:10]:
```suggestion
Note: compared to the Java source, the Kotlin source has
```

**Review by sukolsak — APPROVED** [review]

**Pull request merged**